src/main/java/org/akaza/openclinica/controller/OdmImportController.java

```diff
@@ -29,85 +41,110 @@
             new PassiveExpiringMap<>(24, TimeUnit.HOURS);
 
-
     public OdmImportController(OdmImportService odmImportService, StudyDao studyDao) {
-		super();
-		this.odmImportService = odmImportService;
-		this.studyDao = studyDao;
-	}
+        super();
+        this.odmImportService = odmImportService;
+        this.studyDao = studyDao;
+    }
 
     @SuppressWarnings("unchecked")
     @RequestMapping(value = "/v1/studyversion/boardId/{boardId}", method = RequestMethod.POST)
     public ResponseEntity<Object> importOdmToOC(@RequestBody PublishDTO publishDTO, @PathVariable("boardId") String boardId, HttpServletRequest request)
             throws Exception {
 
         ODM odm = publishDTO.getOdm();
-        List <Page> pages = publishDTO.getPages();
+        List<Page> pages = publishDTO.getPages();
         Instant start = Instant.now();
         String accessToken = (String) request.getSession().getAttribute("accessToken");
 
         try {
-            Map<String, Object> map = (Map<String, Object>) odmImportService.importOdm(odm, pages, boardId, accessToken);
+            Map<String, Object> map = odmImportService.importOdm(odm, pages, boardId, accessToken);
             Study study = (Study) map.get("study");
             Study publicStudy = studyDao.findPublicStudy(study.getOc_oid());
             odmImportService.updatePublicStudyPublishedFlag(publicStudy);
             odmImportService.setPublishedVersionsInFM(map, accessToken);
             Instant end = Instant.now();
-            logger.info("***** Time execustion for {} method : {}   *****", new Object() {
+            logger.info("***** Time execution for {} method : {}   *****", new Object() {
             }.getClass().getEnclosingMethod().getName(), Duration.between(start, end));
-
             return new ResponseEntity<>(HttpStatus.OK);
         } catch (CustomRuntimeException e) {
             return new ResponseEntity<>(e.getErrList(), HttpStatus.BAD_REQUEST);
         }
-
     }
 
-	protected final Logger logger = LoggerFactory.getLogger(getClass().getName());
-	@RequestMapping(value = "/studies/{studyEnvUuid}/publishJobs", method = RequestMethod.POST)
-	public String importOdmToOC(@RequestBody PublishDTO publishDTO, HttpServletRequest request)
-			throws Exception {
+    protected final Logger logger = LoggerFactory.getLogger(getClass().getName());
+
+    @RequestMapping(value = "/studies/{studyEnvUuid}/publishJobs", method = RequestMethod.POST)
+    public ResponseEntity<String> importOdmToOC(@PathVariable("studyEnvUuid") String studyEnvUuid, @RequestBody PublishDTO publishDTO, HttpServletRequest request)
+            throws Exception {
 
         String accessToken = (String) request.getSession().getAttribute("accessToken");
+        Study study = studyDao.findByStudyEnvUuid(studyEnvUuid);
+        String studyOid = study.getOc_oid();
+        utilService.setSchemaFromStudyOid(studyOid);
+        study = studyDao.findByStudyEnvUuid(studyEnvUuid);
```

Comment:
can you add some comment explaining why you are fetching the study again?

Reply:
yup!